src/modules/flow/iio/nodes.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "iio-gen.h"
+#include "sol-flow-internal.h"
+
+#include <sol-util.h>
+#include <errno.h>
+
+#include <sol-iio.h>
+
+struct gyroscope_data {
+    struct sol_flow_node *node;
+    struct sol_iio_device *device;
+    struct sol_iio_channel *channel_x;
+    struct sol_iio_channel *channel_y;
+    struct sol_iio_channel *channel_z;
+    bool buffer_enabled : 1;
+};
+
+static void
+reader_cb(struct sol_iio_device *device, struct sol_iio_buffer *buffer, void *data)
+{
+    const char errmsg[] = "Could not read channel buffer values";
+    struct gyroscope_data *mdata = data;
+    struct sol_direction_vector out;
+    bool b;
+
+    b = sol_iio_read_buffer_channel_value(mdata->channel_x, buffer, &out.x);
+    if (!b) goto error;
+    b = sol_iio_read_buffer_channel_value(mdata->channel_y, buffer, &out.y);
+    if (!b) goto error;
+    b = sol_iio_read_buffer_channel_value(mdata->channel_z, buffer, &out.z);
+    if (!b) goto error;
+
+    // TODO find appropriate min & max (probably, real min e max outputs)
+    sol_flow_send_direction_vector_packet(mdata->node,
+        SOL_FLOW_NODE_TYPE_IIO_GYROSCOPE__OUT__OUT, &out);
+
+    return;
+
+error:
+    sol_flow_send_error_packet(mdata->node, EIO, errmsg);
+    SOL_WRN(errmsg);
```

Comment:
Probably not, as the compiler will most likely generate the same code if it were `printf("%s", foo)` or `printf(foo)`, considering `static const char *foo="bar"`. Since `foo` does not contain any formatting characters, it would be OK.

However, it's basically free to be on the safe side: it's more about the practice of doing things correctly all the time and not thing about exceptions rather than thing about the exceptions all the time.